tests/Plugins/GitHub_EndToEndTest.py

```diff
@@ -0,0 +1,145 @@
+"""End-to-end tests for the GitHub plugin"""
+
+import os
+import sys
+import textwrap
+
+from pathlib import Path
+
+import pytest
+
+from dbrownell_Common.Streams.Capabilities import Capabilities
+from dbrownell_Common.TestHelpers.StreamTestHelpers import (
+    InitializeStreamCapabilities,
+    ScrubDuration,
+)
+from typer.testing import CliRunner
+
+from RepoAuditor.EntryPoint import app
+
+# ----------------------------------------------------------------------
+_github_pat_filename = (Path(__file__).parent / "github_pat.txt").resolve()
+
+if not _github_pat_filename.is_file():
+    sys.stdout.write(
+        textwrap.dedent(
+            f"""\
+
+
+            The filename '{_github_pat_filename}' does not exist. Please create this file and add your GitHub Personal Access Token (PAT) to it.
+            This git repository is configured to ignore the file so that it will never be included as part of a commit.
+
+            To create a new token:
+
+                1. Visit https://github.com/settings/tokens/new
+                2. Ensure that 'repo' scope is checked
+                3. Click 'Generate token'
+                4. Copy the token to the clipboard
+                5. Create the file '{_github_pat_filename}'
+                6. Paste the token into the created file
+                7. Save the file and run these tests again.
+
+            These tests query a repository on GitHub, but GitHub limits the number of concurrent requests made to a repository when a PAT is
+            not provided. As a result, the tests will fail once the limit is reached.
+            """,
+        ),
+    )
+
+    sys.exit(-1)
+
+
+# ----------------------------------------------------------------------
+pytest.fixture(InitializeStreamCapabilities(), scope="session", autouse=True)
+
+
+# ----------------------------------------------------------------------
+def test_Successful(pat_args, snapshot):
+    result = CliRunner().invoke(app, pat_args)
+
+    assert result.exit_code == 0, result.output
+    assert ScrubDuration(result.stdout) == snapshot
+
+
+# ----------------------------------------------------------------------
+def _skip_test_NoPAT():
+    operating_system = os.getenv("GITHUB_CI_OPERATING_SYSTEM")
+    python_version = os.getenv("GITHUB_CI_PYTHON_VERSION")
+
+    if operating_system is None or python_version is None:
+        # Always run locally
+        return False
+
+    # The selection of these values is arbitrary and serve only to ensure that the test is run on
+    # one configuration.
+    return operating_system != "ubuntu-latest" or python_version != "3.12"
+
+
+@pytest.mark.skipif(
+    _skip_test_NoPAT(),
+    reason="Only run this test on the CI machine in specific configurations to avoid GitHub's request throttling.",
+)
+def test_NoPAT(snapshot, args):
+    result = CliRunner().invoke(app, args)
+
+    assert result.exit_code == 1, result.output

```

Comment:
Should this be `result.exit_code == -1` instead of `== 1`?

Reply:
In this case, the exit code is a warning (positive code value) rather than an error (negative code value).